### PR TITLE
BRG-32: Basic scanning for @Inject

### PR DESCRIPTION
**Pull request by azarot**

### CODE REVIEW

File: src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java [hunk 1]
```diff
@@ -33,4 +40,69 @@ public Map<String, BeanDefinition> scan(String packageName) {
...
+    @Override
+    public void fillDependsOn(Map<String, BeanDefinition> beanDefinitions) {
+        for (BeanDefinition beanDefinition : beanDefinitions.values()) {
+            Class<?> beanClass = beanDefinition.getBeanClass();
+            if (beanClass == null) {
+                continue;
+            }
+
+            beanDefinition.setDependsOn(findDependsOn(beanDefinitions, beanClass));
+        }
```
Comment (khshanovskyi) [inline src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java:53]: please, use stream api, smth like:
`     beanDefinitions.values().stream()
          .filter(beanDefinition -> Objects.nonNull(beanDefinition.getBeanClass()))
          .forEach(beanDefinition -> beanDefinition.setDependsOn(findDependsOn(beanDefinitions, beanDefinition.getBeanClass())));`

**Review by NikolayProkopchuk — COMMENTED** [review]
method fillDependsOn is not implemented yet in DefaultBeanScanner, if it will be implemented in next ticket then everything is fine

File: src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java [hunk 2]
```diff
@@ -33,4 +40,69 @@ public Map<String, BeanDefinition> scan(String packageName) {
...
+            }
+
+            String beanName = field.getAnnotation(Inject.class).value();
+            if (StringUtils.isNotEmpty(beanName)) {
+                findDepnesOnByName(beanDefinitions, dependsOn, beanName);
+            } else {
+                findDependsOnByClass(beanDefinitions, beanClass, dependsOn, field);
+            }
+        }
+
```
Comment (khshanovskyi) [inline src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java:70]: can be changed too
`  Arrays.stream(beanClass.getDeclaredFields())
          .filter(field -> field.isAnnotationPresent(Inject.class))
          .forEach(field -> {
              String beanName = field.getAnnotation(Inject.class).value();
              if (StringUtils.isNotEmpty(beanName)) {
                  findDepnesOnByName(beanDefinitions, dependsOn, beanName);
              } else {
                  findDependsOnByClass(beanDefinitions, beanClass, dependsOn, field);
              }
          });`

File: src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java [hunk 3]
```diff
@@ -33,4 +40,69 @@ public Map<String, BeanDefinition> scan(String packageName) {
...
+        return dependsOn.toArray(new String[0]);
+    }
+
+    private static void findDepnesOnByName(Map<String, BeanDefinition> beanDefinitions,
```
Comment (khshanovskyi) [inline src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java:74]: findDepnesOnByName -> findDependsOnByName

File: src/main/java/com/bobocode/svydovets/beans/scanner/DefaultBeanScanner.java [hunk 4]
```diff
@@ -55,7 +55,11 @@ public Map<String, BeanDefinition> scan(String packageName) {
         } catch (IllegalStateException ex) {
             throw new NoUniqueBeanDefinitionException(ex.getMessage());
         }
+    }
 
+    @Override
+    public void fillDependsOn(Map<String, BeanDefinition> beanDefinitions) {
+        //TODO implement
```
[the remark this answers is not in the document]
Comment (azarot) [reply src/main/java/com/bobocode/svydovets/beans/scanner/DefaultBeanScanner.java:62]: Yes, I've just created https://svydovets-bring-framework.atlassian.net/browse/BRN-45?atlOrigin=eyJpIjoiYjBkYzdmYzE4MTllNDNkYzkzYTk2ODU3NjJiODhjOWEiLCJwIjoiaiJ9

**Review by khshanovskyi — APPROVED** [review]

**Review by NikolayProkopchuk — APPROVED** [review]

**Pull request merged**